src/resources/triton_python_backend_utils.py

```diff
@@ -294,5 +294,208 @@ def triton_string_to_numpy(triton_type_string):
     return TRITON_STRING_TO_NUMPY[triton_type_string]
 
 
+def set_max_batch_size(config, value):
+    """Set the max batch size for the model
+    Parameters
+    ----------
+    config : AutoCompleteConfig object
+        object containing the model configuration
+    value : int
+        the max_batch_size we want to set for the model
+    """
+    config.set_max_batch_size(value)
+
+
+def set_input(config, new_input):
+    """Set the input for the model
+    Parameters
+    ----------
+    config : AutoCompleteConfig object
+        object containing the model configuration
+    new_input : InferInputConfig
+        object containing the input we want to set
+        for the model
+    """
+    config.set_input(new_input)
+
+
+def set_output(config, output):
+    """Set the output for the model
+    Parameters
+    ----------
+    config :AutoCompleteConfig object
+        object containing the model configuration
+    output : InferOutputConfig
+        object containing the output we want to set
+        for the model
+    """
+    config.set_output(output)
+
+
+class ModelConfig:
+    """An object of ModelConfig class is used to describe
+    the model configuration for autocomplete.
+    Parameters
+    ----------
+    model_config : dict
+        dictionary object containing the modified model configuration
+    """
+
+    def __init__(self):
+        self._model_config = {}
+        self._model_config["input"] = []
+        self._model_config["output"] = []
+
+    def __str__(self):
+        return str(self._model_config)
+
+    def set_max_batch_size(self, new_max_batch_size):
+        """Set the max batch size for the model.
+        Parameters
+        ----------
+        new_max_batch_size : int
+            The max_batch_size to be set.
+        """
+        self._model_config["max_batch_size"] = new_max_batch_size
+
+    def set_input(self, new_input):
+        """Set the input for the model.
+        Parameters
+        ----------
+        new_input : list
+            The input to be set.
+        """
+        self._model_config["input"].append(new_input._get_input())
+
+    def set_output(self, new_output):
+        """Set the output for the model.
+        Parameters
+        ----------
+        new_output : list
+            The output to be set.
+        """
+        self._model_config["output"].append(new_output._get_output())
```

Comment:
After the discussions we had, we can support set_raw_json too.

Reply:
@krishung5 Please ignore this comment for now. We can add this later when there is a specific request for it.